library-core/src/main/java/com/mikepenz/fastadapter/FastAdapter.java

```diff
@@ -202,8 +202,10 @@ public void setTypeInstanceCache(ITypeInstanceCache<Item> mTypeInstanceCache) {
      */
     public <A extends IAdapter<Item>> FastAdapter<Item> addAdapter(int index, A adapter) {
         mAdapters.add(index, adapter);
+        adapter.withFastAdapter(this);
+        adapter.mapPossibleTypes(adapter.getAdapterItems());
         for (int i = 0; i < mAdapters.size(); i++) {
-            mAdapters.get(i).withFastAdapter(this).setOrder(i);
```

Comment:
the other adapters are already added to the fast adapter so there is no need to call the setter multiple times